pygfx/renderers/wgpu/shaders/backgroundshader.py

```diff
@@ -170,6 +170,14 @@ def code_fragment(self):
             // written in the opaque pass in order for it to really be background.
             // So we fool the blender into thinking this fragment is opaque, even if its not.
             var out = get_fragment_output(varyings.position.z, vec4<f32>(out_color.rgb, 1.0));
+            $$ if write_pick
+            out.pick = (
+                // For a background, it doesn't make sense to report back
+                // anything other than the background id.
+                pick_pack(u32(u_wobject.id), 20)
+                // Maybe this can be different for for textured backgrounds?
```

Comment:
How about including the texcoords like in image shader?

https://github.com/pygfx/pygfx/blob/f163bc82a2d2b244d5aac4500143b7a0f76b34d3/pygfx/renderers/wgpu/shaders/imageshader.py#L258

Reply:
I am happy to look into if you are OK with the general concept. 

The one feature I have the least experience with is the skybox background.

Reply:
+1 on the generic change to make backgrounds pickable.

The background can be a gradient, an image, or a cube/skybox.  Let's at least produce picking info that's consistent. Some options that come to mind:
* Only wobject id. Fine by me. We'll see if we need more later.
* Return 2D screen location, either in NDC or logical pixels.
* Return "texture coordinates": x y, z.

I think I like the latter most. Because a user can easily map the xy texture coordinates to NDC or logical pixels. And this way one can e.g. have a skybox and know what face the user clicked.